src/Castle.Windsor/MicroKernel/Lifestyle/Scoped/CallContextLifetimeScope.cs

```diff
@@ -38,7 +42,11 @@ public class CallContextLifetimeScope : ILifetimeScope
 		private static readonly ConcurrentDictionary<Guid, CallContextLifetimeScope> appDomainLocalInstanceCache =
 			new ConcurrentDictionary<Guid, CallContextLifetimeScope>();
 
+#if FEATURE_REMOTING
 		private static readonly string keyInCallContext = "castle.lifetime-scope-" + AppDomain.CurrentDomain.Id.ToString(CultureInfo.InvariantCulture);
+#else
+		private static readonly AsyncLocal<Guid> CallContextData = new AsyncLocal<Guid>();
```

Comment:
Oh, I was just making an off handed remark about it not existing until .NET 4.6, but that's fine to have both. We'll need to get our README updated soon with descriptions of these defines like Castle Core, can be another PR.